### PR TITLE
[Compute] ScaleInPolicy and PublicIPAddressVersion

### PR DESCRIPTION
**Pull request by hyonholee**
- Add PublicIPAddressVersion parameter to New-AzVmssIPConfig cmdlet

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/368

- Add ScaleInPolicy to New-AzVmss, New-AzVmssConfig and Update-AzVmss cmdlets.

https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/408

- FileUris are moved to protected settings for custom script extension.

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs [hunk 1]
```diff
@@ -519,6 +531,15 @@ private void Run()
                 }
             }
 
+            if (this.IsParameterBound(c => c.ScaleInPolicy))
+            {
+                if (vScaleInPolicy == null)
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs]: I thinks this is always true.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssIpConfigCommand.cs [hunk 2]
```diff
@@ -107,6 +108,12 @@ public partial class NewAzureRmVmssIpConfigCommand : Microsoft.Azure.Commands.Re
             ValueFromPipelineByPropertyName = true)]
         public string PublicIPPrefix { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        [PSArgumentCompleter("IPv4", "IPv6")]
```
Comment (wyunchi-ms) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssIpConfigCommand.cs]: It seems that here missed the default value: `IPv4`

Comment (hyonholee) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssIpConfigCommand.cs]: This is optional.  If a user does not give this parameter, the server sets it as IPv4.

**Comment by wyunchi-ms** [comment]
Hi @hyonholee , could you please take a look?

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs (hunk 1 above)
Comment (hyonholee) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs]: This is from generated code.  The reason for null checking is if there are multiple properties under the same object, we don't want to re-create the object.

**Comment by hyonholee** [comment]
@wyunchi-ms  I replied the comments.

**Pull request merged**